lib/commands/sdk.js

```diff
@@ -549,427 +573,604 @@ SdkSubcommands.install = {
-						var manifest = require(manifestFile);
-						version = manifest.version || version;
-					}
+				if (bar) {
+					bar.tick(total);
+					this.logger.log('\n');
+				}
 
-					callback(version);
+				// get the actual version number
+				const manifestFile = path.join(installLocation, 'manifest.json');
+				if (fs.existsSync(manifestFile)) {
+					const manifest = fs.readJSONSync(manifestFile);
+					this.version = manifest.version || this.version;
 				}
+
+				resolve();
 			});
-		}
+		});
+	}
 
-		async.series([
-			function (next) {
-				if (version) {
-					var versionPath;
-					if (/^http(s)?:\/\/.+/.test(version)) {
-						// version is a URL
-						next(null, { url: version });
-					} else if (/.+\.zip$/.test(version) && fs.existsSync(versionPath = afs.resolvePath(version)) && fs.statSync(versionPath).isFile()) {
-						// version is a file
-						next(null, { file: versionPath });
-					} else {
-						var match = version.match(/^([A-Za-z0-9_]+?):(.+)$/);
-						if (match) {
-							// version is a git hash
-							branch = match[1];
-							doBranch(match[2], next);
-						} else if (branch) {
-							doBranch(version, next);
-						} else {
-							doReleases(next);
-						}
-					}
-				} else if (branch) {
-					doBranch(version, next);
-				} else {
-					doReleases(next);
-				}
-			}
-		], function (err, data) {
-			data = data.shift();
-
-			var vers = Object.keys(cli.env.sdks).filter(function (v) {
-					return appc.version.gte(v, '3.0.0');
-				}).sort().reverse(),
-				newestInstalledSdk = config.get('sdk.selected', config.get('app.sdk', 'latest')) === 'latest' ? (vers.length ? vers[0] : null) : config.get('sdk.selected', config.get('app.sdk')),
-				analyticsPayload = {};
-
-			data.version && (analyticsPayload.version = data.version);
-			data.branch && (analyticsPayload.branch = data.branch);
-
-			function finish(version) {
-				analyticsPayload.version = version;
-				cli.addAnalyticsEvent('sdk.install', analyticsPayload);
-
-				if (version) {
-					var setDefault = (data.setDefault || cli.argv.default) && appc.version.gte(version, '3.0.0') && (cli.argv.default || newestInstalledSdk === null || appc.version.gt(version, newestInstalledSdk));
-					if (setDefault) {
-						logger.log(__('Setting Titanium SDK %s as the default.', version.cyan) + '\n');
-						config.load();
-						config.set('sdk.selected', version);
-						config.save();
-					}
+	/**
+	 * @param {object} sdks
+	 * @returns {Promise<boolean>}
+	 */
+	async isInstalled(sdks) {
+		return Object.hasOwnProperty.call(sdks, this.version);
+	}
+}
 
-					logger.log(__('Titanium SDK %s successfully installed!', version.cyan) + '\n');
-				} else {
-					logger.log(__('Successfully installed!') + '\n');
-				}
+/**
+ * Override for local zipfiles to no-op the download, not delete the zipfile
+ */
+class LocalSDKInstallRequest extends SDKInstallRequest {
+	/**
+	 * @param {Object} logger 
+	 * @param {Object} config 
+	 * @param {CLI} cli 
```

Comment:
<!--
  1 failure:  lib/commands/sdk....
  1 warning:  lib/commands/sdk....
  
  
  DangerID: danger-id-Danger;
  File: lib/commands/sdk.js;
  Line: 1149;
-->
- :no_entry_sign: lib/commands/sdk.js line 1149 – Trailing spaces not allowed. (no-trailing-spaces)
- :warning: lib/commands/sdk.js line 1149 – Missing JSDoc parameter description for 'cli'. (valid-jsdoc)